src/PureScript/Backend/Optimizer/Convert.purs

```diff
@@ -602,10 +611,32 @@ binderToPattern = case _ of
     , patternCase
     , subterms: args # mapWithIndex \idx nextArg ->
         { accessor: buildAccessor idx nextArg
-        , pattern: binderToPattern $ toBinder nextArg
+        , pattern: binderToPattern dataTypes implementations $ toBinder nextArg
         }
     }
 
+  lookupCtorFields
+    :: Qualified ProperName
+    -> Qualified Ident
+    -> Array String
+  lookupCtorFields ty ctor =
+    case importedCtorFields <|> localCtorFields of
+      Just fields -> fields
+      Nothing -> unsafeCrashWith $ "Unable to get fields for type [" <> printQP ty <> "]'s constructor [" <> printQI ctor <> "]"
+    where
+    printQ :: forall a. (a -> String) -> Qualified a -> String
+    printQ printS (Qualified mbMod s) = maybe (printS s) (\(ModuleName mn) -> mn <> "." <> printS s) mbMod
+    printQP = printQ (\(ProperName s) -> s)
+    printQI = printQ (\(Ident i) -> i)
```

Comment:
I don't think printing everything here is necessary. As an assertion, a unique string is enough to locate this and add tracing if it ever arises.

Reply:
This has been dropped.